packages/utils/telemetry-utils/src/telemetryEventBatcher.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+type TCustomData = Record<string, number>;
+
+/**
+ * Telemetry class that accumulates user defined telemetry metrics {@link ICustomDataMap} and sends it to the {@link  ITelemetryLoggerExt} logger provided to this class every time the {@link TelemetryEventBatcher.log} function is called reaches a number specified by the `threshold` value to this classes' constructor.
+ */
+export class TelemetryEventBatcher<TMetrics extends TCustomData> {
+	// Stores value of the custom data passed into the logger.
+	private readonly customDataMap: Map<string, number> = new Map<string, number>();
+	// Counter to keep track of the number of times the log function is called.
+	private counter = 0;
+
+	public constructor(
+		/**
+		 * Custom properties to include in the telemetry performance event when it is written.
+		 */
+		private readonly eventBase: ITelemetryGenericEventExt,
+
+		/**
+		 * The logger to use to write the telemetry performance event.
+		 */
+		private readonly logger: ITelemetryLoggerExt,
+
+		/**
+		 * The number of logs to accumulate before sending the data to the logger.
+		 */
+		private readonly threshold: number,
+	) {}
+
+	/**
+	 * Accumulates the custom data and sends it to the logger every time the number of logs reaches the threshold by calling `sendData()`.
```

Comment:
As a rule, we shouldn't reference private members in public docs.
```suggestion
	 * Accumulates the custom data and sends it to the logger every {@link TelemetryEventBatcher.threshold} calls.
```